### ISSUE TITLE
Segmentation faults

### ISSUE THREAD

**Issue by meiers** [GregoryFaust/yaha#1]
Hey,
I was hoping to use YAHA (version downloaded today) on PacBio reads, but I cannot make it run without segmentation faults. Do you have a clue what the reason could be? Here are the commands I use:

```
yaha -g 1_92130537-92133743.fa -L 12 -H 20
yaha -x 1_92130537-92133743.X12_01_00020S -q reads.fq -P 0.8 -o8 stdout
```

```
YAHA version 0.1.83
WARNING: Index file made with maxHits of 20, while 650 specified for this query run.
Mimimum of two (20) will be used.
r30/127814/0_2743       1:92130537-92133743     80.00   776     37      70      391     1150    2885    2164    -       193     2743    24.39
r30/127814/0_2743       1:92130537-92133743     80.00   102     1       4       1657    1758    1609    1706    +       71      2743    3.54
r30/127814/0_2743       1:92130537-92133743     80.00   257     14      25      1918    2163    1852    2094    +       56      2743    7.95
r30/127814/0_2743       1:92130537-92133743     80.00   361     10      30      2170    2524    1280    944     -       116     2743    11.70
r30/127814/0_2743       1:92130537-92133743     80.00   60      0       2       2647    2705    810     752     -       44      2743    2.11
Segmentation fault
```

I could send you the files if you like to test it, but they standard PacBio reads.

Best,
Sascha

**Comment by GregoryFaust** [comment]
Of course, yaha should not segfault under any conditions.  But a few things stand out to me.  First, what are you using as a reference genome?  Second, how long are the reads?  It looks like it is blowing up on the first or second read, so perhaps you could send me those, and give me some idea where to find the reference you are using.  Finally, using a maxHits (-H) of 20 while making the reference is a very unusual choice, as most of the k-mers will be thrown away unless your reference is very short.

Thanks,
Greg

**Comment by meiers** [comment]
Hi Greg,

thanks for your quick reply. 
I used a very small region of the genome as a reference, since I know reads map to this region. That is why I can use shorter k-mers and a low maxHits filter.
I will attach the files to this email, but they might be rejected while this is going through github. Let’s see..

Best,
Sascha

**Comment by GregoryFaust** [comment]
I didn't get them.  Please try emailing them directly to my email address instead of going through github.
